### DISCUSSION TITLE
make build on raspberry pi return "Error 1"

### DISCUSSION THREAD

**Discussion by SLCCL** [0xERR0R/blocky#398] [Q&A] [answered]
Hello guys, trying to install "Blocky" on my raspberry pi zero 2 W.
The terminal return "Error 1" after I try to "make build". Please give me a hand.

terminal pic shown below:
[https://ibb.co/JdYdzhj](url)

**Comment by SLCCL** [comment]
[https://postimg.cc/VdKLg33r](url)

re-upload the screenshot

**Comment by 0xERR0R** [comment]
Which go version do you use (please execute "go version")?

**Reply by SLCCL** [reply]
pi@blocky-pi:~ $ go version
go version go1.15.15 linux/arm

Thank You!

**Comment by 0xERR0R** [comment, marked as answer]
Try also to execute "go mod tidy"

**Reply by SLCCL** [reply]
pi@blocky-pi:~ $ go mod tidy
go: cannot find main module; see 'go help modules'

Thanks Again!

**Reply by 0xERR0R** [reply]
Please execute it in blocky's directory (where go.mod is located)

**Reply by SLCCL** [reply]
pi@blocky-pi:~/blocky $ go mod tidy
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.15

**Reply by 0xERR0R** [reply]
You need golang 1.17 to build blocky. I think you must install it manually, the version in the official repository of raspbian is too old. You can alternatively consider to use docker, no golang is needed to build the image
